Pspcl/Pspcl.Web/wwwroot/js/stockView.js

```diff
@@ -220,58 +219,51 @@ function validateSerialNumbers(listOfSerialNumber) {
-//        var materialTypeId = $("#materialTypeId").val();
-//        var materialId = $(this).val();
-//        $("#AvailableStock").val('');
-//        if (materialId) {
-//            $.ajax({
-//                url: "/IssueStock/GetAvailableStockRows",
-//                type: "GET",
-//                data: { materialGroupId: materialGroupId, materialTypeId: materialTypeId, materialId: materialId },
-//                success: function (result) {
-//                    if (parseInt(result) > 0)
-//                        $("#AvailableStock").text(result).val(result);
-//                    else {
-//                        showModal('Stock not avaialable.', 'Error..!')
-//                    }
-//                }
-//            });
-//        }
-//    });
-//});   
-
-//$(function () {
-//    $("#materialId").on("change", function () {
-//        var materialGroupId = $("#materialGroupId").val();
-//        var materialTypeId = $("#materialTypeId").val();
-//        var materialId = $(this).val();
-//        $("#makeId").empty();
-//        if (materialId) {
-//            $.ajax({
-//                url: "/IssueStock/GetAllMakes",
-//                type: "GET",
-//                data: { materialGroupId: materialGroupId, materialTypeId: materialTypeId, materialId: materialId },
-//                success: function (result) {
-//                    $("#makeId").append($('<option>').text("--Select Make--").val(""));
-//                    $.each(result, function (i, response) {
-//                        if (response == "") {
-//                            $("#makeId").append($('<option>').text("None").val(response));
-//                        } else {
-//                            $("#makeId").append($('<option>').text(response).val(response));
-//                        }
-//                    });
-//                }
-//            });
-//        }
-//        else {
-//            $("#makeId").append($('<option>').text("--Select Make--").val(""));
-//        }
-//    });
-//});
+$(function () {
+    $("#materialId").on("change", function () {
+        var materialGroupId = $("#materialGroupId").val();
+        var materialTypeId = $("#materialTypeId").val();
+        var materialId = $(this).val();
+        $("#AvailableStock").val('');
+        
+
+        if (materialId) {
+            $.ajax({
+                url: "/IssueStock/DisplayMakeWithQuantity",
+                type: "GET",
+                data: { materialGroupId: materialGroupId, materialTypeId: materialTypeId, materialId: materialId },
+                success: function (response) {
+                    $('#issueMaterialTableBody').empty();
+
+                    var keys = Object.keys(response);
+                    if (keys.length > 0) {
+                        
+                        $('#issueMaterial').show();
+
+                        for (var i = 0; i < keys.length; i++) {
+                            var key = keys[i];
```

Comment:
Please add null check all over the places, where there is a possibility we see that data can be null or empty.